lib/util.js

```diff
@@ -24,11 +24,37 @@ function normalizeVersions (versions) {
 }
 
 function bufferizeVersion (version) {
+  if (typeof version !== 'string' && typeof version !== 'number')
+    throw new Error('invalid version type.')
+
+  if (typeof version === 'string') {
+    return hexStringToBuffer(version)
+  } else if (typeof version === 'number') {
+    // expects BE uint
+    return beUIntToBuffer(version)
+  }
+}
+
+function isValidHexString (input) {
+  var re = /^(0x)?([\dA-Fa-f]{2})+$/g
+  return re.test(input)
+}
+
+function hexStringToBuffer (input) {
+  if (!isValidHexString(input)) throw new Error('invalid hex string.')
+  // omit 0x for buffer
+  var re = /([\dA-Fa-f]{2})+$/g
+  const sanitizedInput = input.match(re)[0]
```

Comment:
You do not need regexp for removing `0x`.

```js
var isValidRE = /^(0x)?([\dA-Fa-f]{2})+$/g
if (!isValidRE.text(input)) throw new Error('invalid hex string.')
return new Buffer(input.slice(2), 'hex')
```

Reply:
Used regexp to provide the option of providing the hex sequence as either "0xdeadbeef" or "deadbeef". If you feel "0x" prefix should be forced, then that change could be made.

Reply:
```js
return new Buffer(input.slice(input.slice(0, 2) == '0x' ? 2 : 0), 'hex')
```
:)

Reply:
I will change to that :). Just wondering for the reason to omit using regexp, as I tend to use it when the opportunity presents itself for formatting/sanitizing user input, based on years old recommendations?